### PR TITLE
Parser: support ** (exponentiation) binary operator (BT-414)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

Add support for the `**` (exponentiation) binary operator to the Beamtalk compiler.

**Linear issue:** https://linear.app/beamtalk/issue/BT-414

### Parser
- Add `**` to `binary_binding_power()` with right-associative precedence (level 50, higher than `*`/`/` at 40)
- Remove `#[allow(dead_code)]` from `right_assoc()` — now used
- Update precedence table documentation

### Codegen
- Add `generate_power_op()` in `operators.rs` — generates `round(math:pow(float(L), float(R)))`
- Matches existing `power_bif` in `primitive_implementations.rs` for consistency

### Tests
- 1 lexer test: `**` tokenized as `BinarySelector("**")`
- 3 parser unit tests: basic parsing, precedence vs `*`, right-associativity
- 7 stdlib tests: basic exponentiation, edge cases (`0 ** 0`, negative base), precedence

### Documentation
- Update `docs/beamtalk-language-features.md` with `**` operator

## Acceptance Criteria
- [x] Lexer tokenises `**` as a single binary operator token
- [x] Parser handles `**` with correct precedence (higher than `*` and `/`)
- [x] Codegen emits the correct primitive call
- [x] Runtime implements `**` dispatch (via existing `@primitive '**'` in Integer.bt)
- [x] Tests pass: `2 ** 10 // => 1024`, `5 ** 0 // => 1`, `7 ** 1 // => 7`

## Release Notes

* **New Features**
  * Added support for the exponentiation operator (**) for power operations.
  * The operator is right-associative with the highest precedence among binary operators, ensuring expressions like `2 ** 3 ** 2` evaluate as `2 ** (3 ** 2)` and `3 * 2 ** 4` evaluates as `3 * (2 ** 4)`.

### CODE REVIEW

File: crates/beamtalk-core/src/codegen/core_erlang/operators.rs [hunk 1]
```diff
@@ -77,6 +82,26 @@ impl CoreErlangGenerator {
...
+    ) -> Result<Document<'static>> {
+        let left_code = self.expression_doc(left)?;
+        let right_code = self.expression_doc(right)?;
+        Ok(docvec![
+            "call 'erlang':'round'(call 'math':'pow'(call 'erlang':'float'(",
+            left_code,
+            "), call 'erlang':'float'(",
+            right_code,
+            ")))",
+        ])
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-core/src/codegen/core_erlang/operators.rs]: Acknowledged — added a doc comment noting the float precision limitation for very large exponents. This is consistent with the existing power_bif in primitive_implementations.rs and is not a regression. A future improvement could use repeated multiplication for exact arbitrary-precision integer results.

File: crates/beamtalk-core/src/codegen/core_erlang/operators.rs [hunk 2]
```diff
@@ -77,6 +82,30 @@ impl CoreErlangGenerator {
...
+    }
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-core/src/codegen/core_erlang/operators.rs]: This is out of scope for BT-414. No other binary operator (`+`, `-`, `*`, `/`, `%`) performs input validation at the codegen level — they all emit direct Erlang BIF calls and let Erlang's runtime type system handle errors (e.g., `badarith` for type mismatches). Adding validation to `**` alone would be inconsistent with the rest of the operator codegen.

The existing `power_bif` in `primitive_implementations.rs` (which handles the `@primitive` path) also has no validation, so the two codegen paths are consistent.

If input validation for arithmetic operators is desired, it should be addressed as a cross-cutting concern for all operators in a follow-up issue.
